Lib/multiprocessing/semaphore_tracker.py

```diff
@@ -68,7 +77,19 @@ def ensure_running(self):
                 exe = spawn.get_executable()
                 args = [exe] + util._args_from_interpreter_flags()
                 args += ['-c', cmd % r]
-                pid = util.spawnv_passfds(exe, args, fds_to_pass)
+                # Register a signal mask that will block the signals. This signal
+                # mask will be inherited by the child that is going to be spawned
+                # and will protect the child from a race condition that can make
+                # the child die before it registers signal handlers for SIGINT and
+                # SIGTERM. The mask is unregistered after spawning the child.
+                # For more info, see: https://bugs.python.org/issue33613
+                if _HAVE_SIGMASK:
```

Comment:
Hum. I would prefer to put the pthread_sigmask() *inside* the try block, to make sure that we always unblock, even if the code is interrupted between pthread_sigmask() and the start of the try block.

Unblocking a signal which isn't blocked does nothing and so is safe.

Reply:
Done in f7d2d4d